### PR TITLE
07 post /api/articles/article id/comments

### PR DESCRIPTION
**Pull request by RogerioSoares96**

### CODE REVIEW

File: controllers.js [hunk 1]
```diff
@@ -34,4 +39,22 @@ exports.getSpecificCommentsByArticleId = (req, res, next) => {
...
+}
+
+exports.postCommentByArticleId = (req, res, next) => {
+    const commentBody = req.body.body;
+    const commentAuthor = req.body.username;
+    checkIfArticleIdExists(req.params.article_id)
+    .then(({article_id}) => {
+        addCommentByArticleId(article_id, commentBody, commentAuthor)
+        .then((returnVal) => {
+            return res.status(201).send({ comment: returnVal })
```
Comment (lewispricey) [inline controllers.js:51]: Great choice of key to return the comment on, nice an descriptive for when we come to use this on the front end 👍

File: models.js [hunk 2]
```diff
@@ -58,4 +58,20 @@ exports.checkIfArticleIdExists = (articleId) => {
...
+};
+
+exports.addCommentByArticleId = (articleId, commentBody, commentAuthor) => {
+    queryParams = [];
+    queryParams.push(articleId)
+    queryParams.push(commentBody)
+    queryParams.push(commentAuthor)
+    return db
+    .query(`INSERT INTO comments (article_id, body, author) 
+            VALUES ($1, $2, $3) RETURNING *;`, queryParams)
```
Comment (lewispricey) [inline models.js:70]: Great to use parameterized query here to prevent any SQL injection

File: app.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const { getAllTopics, getAllArticlesWithCommentCount, getSpecificArticleById, getSpecificCommentsByArticleId} = require('./controllers')
+const { getAllTopics, getAllArticlesWithCommentCount, getSpecificArticleById, getSpecificCommentsByArticleId, postCommentByArticleId} = require('./controllers')
```
Comment (lewispricey) [inline app.js:1]: Awesome to keep you've kept with the same naming convention in your controllers, It makes code super easy to read!

**Review by lewispricey — APPROVED** [review]
Awesome work on this Ticket Rogerio! I've left some individual comments on the files :)

Your testing is great so far however it would be good to add the two following tests:
400 - if the body the user sends is missing a key (ie missing username or body)
201 - if the user passes in an extra key such as votes that this is ignored and defaults to 0.

Please add in those two tests and you're good to go!

**Review by robert112233 — APPROVED** [review]
Great additional tests well done, merge!

**Pull request merged**